apps/mobile/src/libs/components/text/text.tsx

```diff
@@ -0,0 +1,54 @@
+import React, { ReactNode } from "react";
+import { StyleProp, Text as RNText, TextStyle } from "react-native";
+
+import {
+	BaseColor,
+	FontFamilies,
+	FontSize,
+	LineHeight,
+	WeightMap,
+} from "~/libs/enums/enums";
+
+import { styles as presetStyles } from "./styles";
+
+type Properties = {
+	children: ReactNode;
+	color?: keyof typeof BaseColor;
+	fontFamily: keyof typeof FontFamilies;
+	preset?: "default" | "heading" | "subheading";
+	size?: keyof typeof FontSize;
+	style?: StyleProp<TextStyle>;
+	weight?: "bold" | "extraBold" | "medium" | "regular" | "semiBold";
+};
+
+const Text = ({
+	children,
+	color = "BLACK",
+	preset: presetProperties = "default",
+	size,
+	style: styleOverride,
+	weight = "regular",
+}: Properties) => {
+	const baseStyle = presetStyles[presetProperties];
+
+	const sizeToStyleMap = {
+		fontFamily: "NUNITO",
+		fontSize: size ? FontSize[size] : undefined,
+		fontWeight: WeightMap[weight] as TextStyle["fontWeight"],
+		lineHeight: size ? LineHeight[size] : undefined,
+	} satisfies TextStyle;
+
+	const colorStyle: TextStyle = {
+		color: BaseColor[color],
+	};
+
+	const styles: StyleProp<TextStyle> = [
+		baseStyle,
+		sizeToStyleMap,
+		colorStyle,
+		styleOverride,
+	];
```

Comment:
```suggestion
	];
	
```

Reply:
Do you want me to add a new line after the semicolon?

Reply:
Done